soroban-env-host/src/host/conversion.rs

```diff
@@ -332,9 +332,6 @@ impl Host {
 
     pub(crate) fn to_host_val(&self, v: &ScVal) -> Result<Val, HostError> {
         let _span = tracy_span!("ScVal to Val");
-        // This is an internal consistency check: this is an internal method and
-        // any caller should have previously rejected non-representable ScVals.
-        self.check_val_representable_scval(&v)?;
```

Comment:
Why is this being removed?

Reply:
I've decided to use this function for general conversions of potentially malformed data. The 'always valid' case is handled by `to_valid_host_val` function.